common/lib/xmodule/xmodule/seq_module.py

```diff
@@ -238,8 +243,78 @@ def student_view(self, context):
 
         fragment.add_content(self.system.render_template("seq_module.html", params))
 
+        self._capture_full_seq_item_metrics(display_items)
+        self._capture_current_unit_metrics(display_items)
+
+        # Get all descendant XBlock types and counts
         return fragment
 
+    def _locations_in_subtree(self, node):
+        """
+        The usage keys for all descendants of an XBlock/XModule as a flat list.
+        """
+        stack = [node]
+        locations = []
+
+        while stack:
+            curr = stack.pop()
+            locations.append(curr.location)
+            if curr.has_children:
+                stack.extend(curr.get_children())
+
+        return locations
+
+    def _capture_basic_metrics(self):
+        """
+        Capture basic information about this sequence in New Relic before
+        rendering it. We do this up here because proctored exam functionality
+        could short circuit rendering after this section.
+        """
```

Comment:
This docstring seems better as a comment on the actual call. Also, "short circuit" would be better as "bypass" for non-EEs.